### ISSUE TITLE
Application hangs after many connections are saved

### ISSUE THREAD

**Issue by Rollyjoger85** [microsoft/azuredatastudio#15333]
Hi,

After adding all of my SQL connections and saving them in various group Azure Data Studio has become extremely slow when you touch anything that might interact with the settings file (that's my assumption at this point)

This is what I'm experiencing now when ever I change a setting or make a connection from a saved server connection, this will go on for 2-3 minutes and Azure Data Studio will utilize quite a lot of CPU and Memory while it's ongoing.
![image](https://user-images.githubusercontent.com/40987871/117109535-1afd5e00-ad85-11eb-98db-7cda3eecf6da.png)
![image](https://user-images.githubusercontent.com/40987871/117110165-0372a500-ad86-11eb-894a-34200468f6c0.png)

It might be a little on the extreme side but at the moment I'm having 2200 saved connections (it's really useful for my job, since I can easily search the list and find the connection I need to work on) hence my settings file has grown quite a lot.

If i remove all my saved connections Azure Data Studio works fine, with absoultely no application hang, I'm in doubt whether this should classify as a bug or feature request, I mean it works but it hangs and it's pretty annoying to the point where it's becoming unusable for daily operation.

Azure Data Studio version: azuredatastudio 1.28.0 (13e362762762e5cb76a5c8afd2bc650f48c2d3d8, 2021-04-15T00:24:15.710Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.85GB (5.48GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
agent|Mic|0.48.0
cms|Mic|0.8.0
dacpac|Mic|1.8.0
import|Mic|1.3.0
managed-instance-dashboard|Mic|0.4.2
profiler|Mic|0.11.0
query-history|Mic|0.1.0
schema-compare|Mic|1.10.0
server-report|Mic|0.2.3
whoisactive|Mic|0.1.4
vscodeintellicode|Vis|1.2.1
gruvbox-concoctis|whe|10.30.27

(1 theme extensions excluded)

</details>

**Comment by cheenamalhotra** [comment]
@rgk85 

Checking in, do you still experience this issue with latest versions of ADS?

**Comment by rasmus-kristensen** [comment]
No, I'm not longer experiencing this issue using the latest version, thanks!
